core/src/apps/management/recovery_device/recover.py

```diff
@@ -57,10 +69,21 @@ def _process_slip39(words: str) -> Optional[bytes]:
         raise RuntimeError("Slip39: This mnemonic was already entered")
 
     # add mnemonic to storage
-    remaining -= 1
-    storage.recovery.set_remaining(remaining)
-    storage.recovery_shares.set(index, words)
-    if remaining != 0:
+    if group_count > 1:
+        # TODO: investigate why shares_remianing is bytes instead of bytearray
+        shares_remaining = bytearray(shares_remaining)
+        shares_remaining[group_index] = shares_remaining[group_index] - 1
```

Comment:
this will save the wrong value in case of share that wasn't entered yet. will be fixed.

Reply:
should be replaced with threshold - 1 like above.